### DISCUSSION TITLE
shotcut: changes to settings do not persist with default.profile (AppImage)

### DISCUSSION THREAD

**Discussion by tredondo** [netblue30/firejail#4199] [Q&A]
I'm using the [Shotcut .appimage](https://shotcut.org/download/). If I launch it normally and change a setting, e.g. uncheck Settings -> Scrub Audio, Shotcut remembers that setting next time I launch it.

But if I launch it with `firejail --appimage ./shotcut-linux-x86_64-210321.AppImage`, the setting reverts to the default (checked) next time I run the same firejail shotcut command.

If I add `--profile=shotcut` to the command line, the settings are preserved.

Shotcut stores its config file in `~/.config/Meltytech/Shotcut.conf` and other files in `~/.local/share/Meltytech/Shotcut`. The instance launched via `firejail --appimage` (without `--profile`) does have access to both. So I don't understand why Shotcut wouldn't be able to save its settings even in that case.

Anyway, [the documentation on appimage support](https://firejail.wordpress.com/documentation-2/appimage-support/) doesn't mention anything about using profiles. Is it recommended to use the application's profile anyway? That should be added to the page.

**Comment by ghost** [comment]
The shotcut.profile has `noblacklist ${HOME}/.config/Meltytech`. Don't you have that dir on your system when using shotcut?

> Anyway, the documentation on appimage support doesn't mention anything about using profiles. Is it recommended to use the application's profile anyway? That should be added to the page.

Correct, that is the intended way to use AppImages in combination with firejail. Without specifying a profile firejail uses `default.profile`, just as it does for apps that don't have a profile. When a specific firejail profile does exist, it would make sense to specify it, for hardened support. But then again, if such a profile existed I would assume there's native packaging support for it available throughout most Linux'es. Firejail was never designed with the idea of there being things like AppImages, Flatpak etc. And that shows.

I can't find any requests/PR's to auto-check and add a specific profile based on the passed file name when --appimage is used. In theory that would be possible. The wordpress documentation is being overhauled (although I'm not sure about the specifics). Until that's finished, a good place to start making this behavior more explicit would be man firejail.
